source/guides/network/running-a-symbol-node-manually.rst

```diff
@@ -0,0 +1,194 @@
+
+.. note:: The following instructions have only been verified to work on Linux.
+
+*****************
+Build a Peer node
+*****************
+
+Build the server
+================
+
+Follow the `Catapult server <https://github.com/nemtech/catapult-server>`__ instructions to **build** its binaries, either `manually <https://github.com/nemtech/catapult-server/blob/main/docs/BUILD-manual.md>`__ or using `Conan <https://github.com/nemtech/catapult-server/blob/main/docs/BUILD-conan.md>`__.
+
+Run the server
+==============
+
+**Download** the network configuration file from **A LOCATION YET TO BE DETERMINED**.
+
+Follow Catapult server's `instructions to run a node <https://github.com/nemtech/catapult-server/blob/main/docs/RUNPEERLIN.md>`__ using the downloaded network configuration.
+
+*****************
+Build an API node
+*****************
+
+Once you have the Catapult Server running, you can add **extra services** to turn it into an API node:
+
+.. figure:: ../../resources/images/diagrams/rest-detail.png
+    :align: center
+
+As shown in the diagram above, besides the **Catapult Server** you will also need:
+
+- A **Database** holding the REST data.
+- A **Catapult Broker** serializing accesses to the database.
+- A **REST gateway** accepting client requests and turning them into server commands or database queries.
+
+The following sections explain how to install each service. You will need all of them.
+
+Move to the ``_build`` folder created while building the server and create any additional folders requested in the next steps inside ``_build``.
+
+Run the database
+=====================
+
+1. **Install** `MongoDB <http://mongodb.com>`__ (at least version 4.4).
+
+   It is recommended to follow the `program's own installation instructions <https://docs.mongodb.com/manual/administration/install-community/>`__, and NOT install from your distro's packages.
+
+2. Create a directory to store the data files and **run the database**:
+
+   .. code-block:: bash
+
+      mkdir dbfiles
+      mongod --dbpath=dbfiles --wiredTigerCacheSizeGb 2 --bind_ip 127.0.0.1
+
+   Keep ``mongod`` running for as long as your node runs.
+
+3. **Build indices** to optimize database accesses.
+
+   The ``catapult-server`` repository you cloned to build the server in the first step contains `a folder with mongo scripts <https://github.com/nemtech/catapult-server/blob/main/scripts/mongo>`__. Run this one:
+
+   .. code-block:: bash
+
+      mongo 127.0.0.1/catapult < ../../scripts/mongo/mongoDbPrepare.js
+
+   Without this step database performance will be unacceptably slow.
+
+Run the broker
+===================
+
+4. **Edit** ``resources/config-database.properties`` and point ``databaseUri`` to ``127.0.0.1:27017``:
+
+   .. code-block:: ini
+
+      [database]
+  
+      databaseUri = mongodb://127.0.0.1:27017
+      databaseName = catapult
+
+5. **Run the broker**:
+
+   Move into the ``bin`` directory and run:
+
+   .. code-block:: bash
+
+      ./catapult.broker ..
```

Comment:
I have a crash here
```
marc-os@roston:~/mm-s/main/catapult-server/_build/bin$ ./catapult.broker ..
Copyright (c) Jaguar0625, gimre, BloodyRookie, Tech Bureau, Corp.
catapult version: 0.10.0.8 ac70cf8e [devnet]
loading resources from "../resources"
loading configuration from "../resources/config-inflation.properties"
loading configuration from "../resources/config-extensions-broker.properties"
loading configuration from "../resources/config-user.properties"
loading configuration from "../resources/config-logging-broker.properties"
loading configuration from "../resources/config-node.properties"
loading configuration from "../resources/config-network.properties"
2021-03-05 21:09:09.475202 0x00007f875c279bc0: <warning> (process::Platform.cpp@39) 
	Current user umask settings are too wide '22'.
	It's strongly suggested to use umask value '077' when starting symbol software. 
2021-03-05 21:09:09.527056 0x00007f875c279bc0: <important> (process::ProcessMain.cpp@90) booting process with keys:
 -   CA EB6003935A7422EAEE04DDD4D7798E076DF672FDA6A3E37A234012C153969230
 - NODE DB5FC52F0F0AF9D8FAF6BF37D4B9FF7CCCA389A59AA34609796576FBF09D255D 
2021-03-05 21:09:09.528391 0x00007f875c279bc0: <info> (thread::IoThreadPool.cpp@105) spawned 8 workers (broker) 
2021-03-05 21:09:09.528717 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libextension.addressextraction.so 
2021-03-05 21:09:09.581109 0x00007f875c279bc0: <info> (extensions::ProcessBootstrapper.cpp@121) registering dynamic extension extension.addressextraction 
2021-03-05 21:09:09.581368 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libextension.mongo.so 
2021-03-05 21:09:09.935788 0x00007f875c279bc0: <info> (extensions::ProcessBootstrapper.cpp@121) registering dynamic extension extension.mongo 
loading configuration from "../resources/config-database.properties"
2021-03-05 21:09:09.973814 0x00007f875c279bc0: <info> (thread::IoThreadPool.cpp@105) spawned 8 workers (bulk writer) 
2021-03-05 21:09:09.985325 0x00007f875c279bc0: <info> (src::MongoPluginLoader.cpp@56) registering dynamic mongo plugin catapult.mongo.plugins.restrictionmosaic 
2021-03-05 21:09:09.985542 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.mongo.plugins.restrictionmosaic.so 
2021-03-05 21:09:09.988909 0x00007f875c279bc0: <info> (src::MongoPluginLoader.cpp@56) registering dynamic mongo plugin catapult.mongo.plugins.restrictionaccount 
2021-03-05 21:09:09.989056 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.mongo.plugins.restrictionaccount.so 
2021-03-05 21:09:09.990440 0x00007f875c279bc0: <info> (src::MongoPluginLoader.cpp@56) registering dynamic mongo plugin catapult.mongo.plugins.multisig 
2021-03-05 21:09:09.990466 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.mongo.plugins.multisig.so 
2021-03-05 21:09:09.991047 0x00007f875c279bc0: <info> (src::MongoPluginLoader.cpp@56) registering dynamic mongo plugin catapult.mongo.plugins.mosaic 
2021-03-05 21:09:09.991075 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.mongo.plugins.mosaic.so 
2021-03-05 21:09:09.991640 0x00007f875c279bc0: <info> (src::MongoPluginLoader.cpp@56) registering dynamic mongo plugin catapult.mongo.plugins.metadata 
2021-03-05 21:09:09.991669 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.mongo.plugins.metadata.so 
2021-03-05 21:09:09.992286 0x00007f875c279bc0: <info> (src::MongoPluginLoader.cpp@56) registering dynamic mongo plugin catapult.mongo.plugins.locksecret 
2021-03-05 21:09:09.992312 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.mongo.plugins.locksecret.so 
2021-03-05 21:09:09.992908 0x00007f875c279bc0: <info> (src::MongoPluginLoader.cpp@56) registering dynamic mongo plugin catapult.mongo.plugins.lockhash 
2021-03-05 21:09:09.992933 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.mongo.plugins.lockhash.so 
2021-03-05 21:09:09.993516 0x00007f875c279bc0: <info> (src::MongoPluginLoader.cpp@56) registering dynamic mongo plugin catapult.mongo.plugins.aggregate 
2021-03-05 21:09:09.993543 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.mongo.plugins.aggregate.so 
2021-03-05 21:09:09.993824 0x00007f875c279bc0: <info> (src::MongoPluginLoader.cpp@56) registering dynamic mongo plugin catapult.mongo.plugins.transfer 
2021-03-05 21:09:09.993849 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.mongo.plugins.transfer.so 
2021-03-05 21:09:09.994105 0x00007f875c279bc0: <info> (src::MongoPluginLoader.cpp@56) registering dynamic mongo plugin catapult.mongo.plugins.namespace 
2021-03-05 21:09:09.994128 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.mongo.plugins.namespace.so 
2021-03-05 21:09:09.994743 0x00007f875c279bc0: <info> (src::MongoPluginLoader.cpp@56) registering dynamic mongo plugin catapult.mongo.plugins.accountlink 
2021-03-05 21:09:09.994768 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.mongo.plugins.accountlink.so 
2021-03-05 21:09:09.995916 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libextension.zeromq.so 
2021-03-05 21:09:10.108912 0x00007f875c279bc0: <info> (extensions::ProcessBootstrapper.cpp@121) registering dynamic extension extension.zeromq 
loading configuration from "../resources/config-messaging.properties"
2021-03-05 21:09:10.173560 0x00007f875c279bc0: <info> (thread::IoThreadPool.cpp@105) spawned 1 workers (ZeroMqEntityPublisher) 
2021-03-05 21:09:10.173808 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libextension.hashcache.so 
2021-03-05 21:09:10.200215 0x00007f875c279bc0: <info> (extensions::ProcessBootstrapper.cpp@121) registering dynamic extension extension.hashcache 
2021-03-05 21:09:10.200519 0x00007f875c279bc0: <info> (broker::Broker.cpp@61) registering system plugins 
2021-03-05 21:09:10.200640 0x00007f875c279bc0: <info> (plugins::PluginLoader.cpp@48) registering dynamic plugin catapult.plugins.coresystem 
2021-03-05 21:09:10.200737 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.plugins.coresystem.so 
2021-03-05 21:09:10.207257 0x00007f875c279bc0: <info> (plugins::PluginLoader.cpp@48) registering dynamic plugin catapult.plugins.signature 
2021-03-05 21:09:10.207371 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.plugins.signature.so 
2021-03-05 21:09:10.209322 0x00007f875c279bc0: <info> (plugins::PluginLoader.cpp@48) registering dynamic plugin catapult.plugins.hashcache 
2021-03-05 21:09:10.209391 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.plugins.hashcache.so 
2021-03-05 21:09:10.211911 0x00007f875c279bc0: <info> (plugins::PluginLoader.cpp@48) registering dynamic plugin catapult.plugins.transfer 
2021-03-05 21:09:10.211986 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.plugins.transfer.so 
2021-03-05 21:09:10.214080 0x00007f875c279bc0: <info> (plugins::PluginLoader.cpp@48) registering dynamic plugin catapult.plugins.namespace 
2021-03-05 21:09:10.214142 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.plugins.namespace.so 
2021-03-05 21:09:10.217431 0x00007f875c279bc0: <info> (plugins::PluginLoader.cpp@48) registering dynamic plugin catapult.plugins.restrictionmosaic 
2021-03-05 21:09:10.217491 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.plugins.restrictionmosaic.so 
2021-03-05 21:09:10.220213 0x00007f875c279bc0: <info> (plugins::PluginLoader.cpp@48) registering dynamic plugin catapult.plugins.metadata 
2021-03-05 21:09:10.220263 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.plugins.metadata.so 
2021-03-05 21:09:10.222537 0x00007f875c279bc0: <info> (plugins::PluginLoader.cpp@48) registering dynamic plugin catapult.plugins.multisig 
2021-03-05 21:09:10.222583 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.plugins.multisig.so 
2021-03-05 21:09:10.224672 0x00007f875c279bc0: <info> (plugins::PluginLoader.cpp@48) registering dynamic plugin catapult.plugins.lockhash 
2021-03-05 21:09:10.224715 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.plugins.lockhash.so 
2021-03-05 21:09:10.226840 0x00007f875c279bc0: <info> (plugins::PluginLoader.cpp@48) registering dynamic plugin catapult.plugins.mosaic 
2021-03-05 21:09:10.226882 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.plugins.mosaic.so 
2021-03-05 21:09:10.228907 0x00007f875c279bc0: <info> (plugins::PluginLoader.cpp@48) registering dynamic plugin catapult.plugins.locksecret 
2021-03-05 21:09:10.228942 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.plugins.locksecret.so 
2021-03-05 21:09:10.230817 0x00007f875c279bc0: <info> (plugins::PluginLoader.cpp@48) registering dynamic plugin catapult.plugins.accountlink 
2021-03-05 21:09:10.230852 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.plugins.accountlink.so 
2021-03-05 21:09:10.232102 0x00007f875c279bc0: <info> (plugins::PluginLoader.cpp@48) registering dynamic plugin catapult.plugins.restrictionaccount 
2021-03-05 21:09:10.232136 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.plugins.restrictionaccount.so 
2021-03-05 21:09:10.234073 0x00007f875c279bc0: <info> (plugins::PluginLoader.cpp@48) registering dynamic plugin catapult.plugins.aggregate 
2021-03-05 21:09:10.234104 0x00007f875c279bc0: <info> (plugins::PluginModule.cpp@83) loading plugin from ./libcatapult.plugins.aggregate.so 
2021-03-05 21:09:10.234974 0x00007f875c279bc0: <info> (utils::StackLogger.h@37) pushing scope 'booting broker' 
2021-03-05 21:09:10.235214 0x00007f875c279bc0: <info> (utils::StackLogger.h@43) popping scope 'booting broker' (0ms) 
2021-03-05 21:09:10.235277 0x00007f875c279bc0: <info> (process::Signals.cpp@45) waiting for termination signal 
2021-03-05 21:09:10.336176 0x00007f875926d700: <error> (io::FileQueue.cpp@133) Throwing exception: reading from file queue failed due to missing message file ("../data/spool/state_change/0000000000000000.dat") 
2021-03-05 21:09:10.340484 0x00007f875926d700: <fatal> (process::ProcessMain.cpp@75) 
thread: 5 (broker) work
unhandled exception while running local node!
../src/catapult/io/FileQueue.cpp(133): Throw in function bool catapult::io::FileQueueReader::process(catapult::consumer<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&>&)
Dynamic exception type: boost::wrapexcept<catapult::catapult_error<std::runtime_error> >
std::exception::what: reading from file queue failed due to missing message file
[catapult::ErrorParam1*] = "../data/spool/state_change/0000000000000000.dat"
 
```

Reply:
```bash
total 24
drwx------ 2 marc-os marc-os 4096 Mar  5 21:16 .
drwx------ 9 marc-os marc-os 4096 Mar  5 21:09 ..
-rw------- 1 marc-os marc-os    8 Mar  5 21:09 index_broker_r.dat
-rw------- 1 marc-os marc-os    8 Mar  5 21:16 index.dat
-rw------- 1 marc-os marc-os    8 Mar  5 21:16 index_server.dat
-rw------- 1 marc-os marc-os    8 Mar  5 21:16 index_server_r.dat

```

Reply:
processes catapult-server and mongodb are running fine at any moment before and after the broker crash

Reply:
I don't know what to say here... did restarting the broker fox the issue? @gimer @Jaguar0625